controllers/securitypolicy_controller.go

```diff
@@ -88,121 +92,160 @@ func (r *SecurityPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 	}
+
 	return spec, nil
 }
 
 func (r *SecurityPolicyReconciler) updateAccess(ctx context.Context,
 	a *tykv1.AccessDefinition) error {
 	api := &tykv1.ApiDefinition{}
+
 	if err := r.Get(ctx, types.NamespacedName{Name: a.Name, Namespace: a.Namespace}, api); err != nil {
 		if errors.IsNotFound(err) {
 			r.Log.Info("ApiDefinition resource not found. Unable to attach to SecurityPolicy. ReQueue",
 				"Name", a.Name,
 				"Namespace", a.Namespace,
 			)
+
 			return err
 		}
+
 		r.Log.Error(err, "Failed to get APIDefinition to attach to SecurityPolicy")
+
 		return err
 	}
+
 	if api.Status.ApiID == "" {
 		return opclient.ErrNotFound
 	}
+
 	a.APIID = api.Status.ApiID
 	a.APIName = api.Spec.Name
+
 	return nil
 }
 
 func (r *SecurityPolicyReconciler) delete(ctx context.Context, policy *tykv1.SecurityPolicy) error {
 	r.Log.Info("Deleting policy")
+
 	all, err := klient.Universal.Portal().Catalogue().Get(ctx)
 	if err != nil {
 		return err
 	}
+
 	for _, v := range all.APIS {
 		if v.PolicyID == policy.Status.PolID {
 			return fmt.Errorf("cannot delete policy due to catalogue %q dependency", all.Id)
 		}
 	}
+
 	util.RemoveFinalizer(policy, policyFinalizer)
+
 	if err := klient.Universal.Portal().Policy().Delete(ctx, policy.Status.PolID); err != nil {
 		if opclient.IsNotFound(err) {
 			r.Log.Info("Policy not found")
 			return nil
 		}
+
 		r.Log.Error(err, "Failed to delete resource")
+
 		return err
 	}
+
 	err = r.updateLinkedAPI(ctx, policy, func(ads *tykv1.ApiDefinitionStatus, ns model.Target) {
 		ads.LinkedByPolicies = removeTarget(ads.LinkedByPolicies, ns)
 	})
 	if err != nil {
 		return err
 	}
+
 	r.Log.Info("Successfully deleted Policy")
+
 	return nil
 }
 
 func (r *SecurityPolicyReconciler) update(ctx context.Context, policy *tykv1.SecurityPolicy) error {
 	r.Log.Info("Updating  policy")
+
 	policy.Spec.MID = policy.Status.PolID
+
 	spec, err := r.spec(ctx, policy)
 	if err != nil {
 		return err
 	}
+
 	err = klient.Universal.Portal().Policy().Update(ctx, spec)
 	if err != nil {
 		r.Log.Error(err, "Failed to update policy")
 		return err
 	}
+
 	err = r.updateLinkedAPI(ctx, policy, func(ads *tykv1.ApiDefinitionStatus, s model.Target) {
 		ads.LinkedByPolicies = addTarget(ads.LinkedByPolicies, s)
 	})
+
 	if err != nil {
 		return err
 	}
+
 	klient.Universal.HotReload(ctx)
 	r.Log.Info("Successfully updated Policy")
+
 	return nil
 }
 
 func (r *SecurityPolicyReconciler) create(ctx context.Context, policy *tykv1.SecurityPolicy) error {
 	r.Log.Info("Creating  policy")
+
 	spec, err := r.spec(ctx, policy)
 	if err != nil {
 		return err
 	}
+
 	err = klient.Universal.Portal().Policy().Create(ctx, spec)
 	if err != nil {
 		r.Log.Error(err, "Failed to create policy")
+
 		return err
 	}
+
 	err = r.updateLinkedAPI(ctx, policy, func(ads *tykv1.ApiDefinitionStatus, s model.Target) {
 		ads.LinkedByPolicies = addTarget(ads.LinkedByPolicies, s)
 	})
+
+	if err != nil {
+		r.Log.Error(err, "failed to update linkedAPI status")
+	}
```

Comment:
If there is an error updating the linked API status - it's OK to continue? If so, why?